automq-shell/src/main/java/com/automq/shell/command/ForceClose.java

```diff
@@ -0,0 +1,138 @@
+        if (controllerOptional.isEmpty()) {
+            System.err.println("No controller node found.");
+            return 1;
+        }
+
+        NetworkClient client = CLIUtils.buildNetworkClient("automq-cli", new AdminClientConfig(new Properties()), new Metrics(), Time.SYSTEM, new LogContext());
+        ClientStreamManager manager = new ClientStreamManager(client, controllerOptional.get());
+
+        List<DescribeStreamsResponseData.StreamMetadata> list;
+        if (exclusive.streamId > 0) {
+            list = manager.describeStreamsByStream(exclusive.streamId);
+        } else if (exclusive.nodeId > 0) {
+            list = manager.describeStreamsByNode(exclusive.nodeId);
+        } else {
+            list = manager.describeStreamsByTopicPartition(Map.of(exclusive.topicPartition.topicName, exclusive.topicPartition.partitionList));
+        }
+
+        if (list.isEmpty()) {
+            System.out.println("No stream found.");
+            return 0;
+        }
+
+        list = list.subList(0, Math.min(maxCount, list.size()));
+
+        if (force) {
+            System.out.println();
+            System.out.println("Force close " + list.size() + " stream(s)...");
+
+            for (DescribeStreamsResponseData.StreamMetadata metadata : list) {
+                if (StreamState.valueOf(metadata.state()) == StreamState.OPENED) {
+                    manager.closeStream(metadata.streamId(), metadata.epoch(), metadata.nodeId());
+                    System.out.println("Node: " + metadata.nodeId() + ", Stream: " + metadata.streamId() + ", TopicPartition: " + metadata.topicName() + "-" + metadata.partitionIndex() + " closed.");
+                } else {
+                    System.out.println("Node: " + metadata.nodeId() + ", Stream: " + metadata.streamId() + ", TopicPartition: " + metadata.topicName() + "-" + metadata.partitionIndex() + " is already closed.");
+                }
+            }
+        } else {
+            String info = CommandLine.Help.Ansi.AUTO.string("@|bold Dry run mode, no stream will be closed.|@");
+            System.out.println(info);
+            System.out.println();
+            System.out.println("Found following stream(s):");
+
+            for (DescribeStreamsResponseData.StreamMetadata metadata : list) {
+                System.out.println("Node: " + metadata.nodeId() + ", Stream: " + metadata.streamId() + ", TopicPartition: " + metadata.topicName() + "-" + metadata.partitionIndex() + ", State: " + StreamState.valueOf(metadata.state()));
+            }
+
+            System.out.println();
+            String warning = CommandLine.Help.Ansi.AUTO.string("@|bold,yellow WARNING: Closing the stream forcefully will result in the loss of uncommitted data. To proceed, please confirm by using --force to force close the stream(s).|@");
+            System.out.println(warning);
+        }
+
+        return 0;
+    }
+
+    private Optional<Node> findAnyControllerNode(Admin admin) throws Exception {
```

Comment:
Adhering to the Kafka standard for other ApiKeys, like ApiKeys.ELECT_LEADERS, when forwarding from Broker proxies to the Controller, would be advisable.

One of the reasons for doing so is that the Controller's network may not necessarily be interconnected with the client.